### PR TITLE
adding ieeg for filtered speech data

### PR DESCRIPTION
**Pull request by choldgraf**
This is a first pass at the filtered speech dataset. I'm probably still missing some stuff / have things incorrectly-named here and there, but it's the bulk of the work. Would love to hear thoughts/comments/etc and I will make another few rounds of edits.

One quirk to this dataset is that the electrode localization data we have is only at the level of x/y positions on a 2-D snapshot. I've made the corresponding naming decisions that seemed reasonable, but perhaps there's a better way to do it? If necessary I can go digging around to see if I can find the nifty files, but if I recall, they were really variable in terms of completeness/quality of the data (since this dataset came from many different recording sites and much of it was collected a really long time ago)

cc @dorahermes @chrisfilo

### CODE REVIEW

File: ieeg_filtered_speech/sub-cm4/ses-01/ieeg/sub-cm4_task-filtered-speech_run-01_channel.tsv [hunk 1]
```diff
@@ -0,0 +1,65 @@
+name	type	description	sampling_frequency	low_cutoff	high_cutoff	status	reference	notch	units
+G1	ECoG	Electrocorticography	1000.0	1.0	499.71	good	common-average-reference	n/a	MicroVolt
```
Comment (dorahermes) [inline ieeg_filtered_speech/sub-cm4/ses-01/ieeg/sub-cm4_task-filtered-speech_run-01_channel.tsv]: ECoG should be ECOG - all caps

Comment (dorahermes) [inline ieeg_filtered_speech/sub-cm4/ses-01/ieeg/sub-cm4_task-filtered-speech_run-01_channel.tsv]: Units should be microV according to BIDS spec for Units and Prefixes

File: ieeg_filtered_speech/sub-cm4/ses-01/ieeg/sub-cm4_task-filtered-speech_run-01_events.tsv [hunk 2]
```diff
@@ -0,0 +1,184 @@
+Onset	duration	filt_type	kind	t_num	t_type
+552.835	0.728428200000053	nan	bad	nan	nan
```
Comment (dorahermes) [inline ieeg_filtered_speech/sub-cm4/ses-01/ieeg/sub-cm4_task-filtered-speech_run-01_events.tsv]: nan should be n/a

File: ieeg_filtered_speech/sub-cm4/ses-01/ieeg/sub-cm4_task-filtered-speech_run-01_events.tsv [hunk 3]
```diff
@@ -0,0 +1,184 @@
+Onset	duration	filt_type	kind	t_num	t_type
```
Comment (dorahermes) [inline ieeg_filtered_speech/sub-cm4/ses-01/ieeg/sub-cm4_task-filtered-speech_run-01_events.tsv]: onset should be without capital, t_type should be trial_type

File: ieeg_filtered_speech/sub-cm4/ses-01/ieeg/sub-cm4_task-filtered-speech_run-01_space-2DImage_coordsystem.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+    "iEEGCoordinateSystem": "2D Image",
```
Comment (dorahermes) [inline ieeg_filtered_speech/sub-cm4/ses-01/ieeg/sub-cm4_task-filtered-speech_run-01_space-2DImage_coordsystem.json]: Let's add a term for this in the spec, without spaces

File: ieeg_filtered_speech/sub-cm4/ses-01/ieeg/sub-cm4_task-filtered-speech_run-01_space-2DImage_coordsystem.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+    "iEEGCoordinateUnits": "n/a",
```
Comment (dorahermes) [inline ieeg_filtered_speech/sub-cm4/ses-01/ieeg/sub-cm4_task-filtered-speech_run-01_space-2DImage_coordsystem.json]: MUST be indices for pictures

File: ieeg_filtered_speech/sub-cm4/ses-01/ieeg/sub-cm4_task-filtered-speech_run-01_space-2DImage_coordsystem.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+    "iEEGCoordinateProcessingDescripton": "Electrodes localized with bioimage suite",
+    "IntendedFor": "/sub-cm4/ses-01/anat/sub-cm4_elec-recon.png",
+    "AssociatedImageCoordinateSystem": "n/a",
```
Comment (dorahermes) [inline ieeg_filtered_speech/sub-cm4/ses-01/ieeg/sub-cm4_task-filtered-speech_run-01_space-2DImage_coordsystem.json]: See spec for what must be used for pictures

File: ieeg_filtered_speech/sub-cm4/ses-01/ieeg/sub-cm4_task-filtered-speech_run-01_space-2DImage_electrodes.tsv [hunk 7]
```diff
@@ -0,0 +1,65 @@
+type	x	y	name
+grid	421.0	85.0	G1
```
Comment (dorahermes) [inline ieeg_filtered_speech/sub-cm4/ses-01/ieeg/sub-cm4_task-filtered-speech_run-01_space-2DImage_electrodes.tsv]: type must be one of surface/depth/dbs and add columns for size of the electrode contacts

File: ieeg_filtered_speech/sub-cm4/ses-01/sub-cm4_ses-01_ieeg.json [hunk 8]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (dorahermes) [inline ieeg_filtered_speech/sub-cm4/ses-01/sub-cm4_ses-01_ieeg.json]: Many missing or incorrectly labeled fields, see spec. Why 64 eeg channels? no iEEGsurf? no iEEGdepth? why MEG/MEGREF, and several others
Add manufacturer

**Comment by dorahermes** [comment]
Great to have another dataset added! I made several notes on the first set of files, but the same applies to other files of the same kind

File: ieeg_filtered_speech/sub-cm8/ses-01/.ipynb_checkpoints/sub-cm8_ses-01_ieeg-checkpoint.json [hunk 9]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (dorahermes) [inline ieeg_filtered_speech/sub-cm8/ses-01/.ipynb_checkpoints/sub-cm8_ses-01_ieeg-checkpoint.json]: what is ieeg-checkpoint?

**Comment by choldgraf** [comment]
these are all super helpful - thanks Dora. One reason for some of the discrepancies is that I'm trying to add iEEG functionality to MNE-BIDS while I'm converting this data, so there are some fields that I've missed that do things slightly differently. I'll update and push a new version of the data soon!

File: ieeg_filtered_speech/sub-cm4/ses-01/ieeg/sub-cm4_task-filtered-speech_run-01_space-2DImage_electrodes.tsv (hunk 7 above)
Comment (choldgraf) [reply ieeg_filtered_speech/sub-cm4/ses-01/ieeg/sub-cm4_task-filtered-speech_run-01_space-2DImage_electrodes.tsv]: good call - need to track this down as it's not reported in my paper (and apparently none of the other knight lab papers I randomly sampled either :-P )

File: ieeg_filtered_speech/sub-cm4/ses-01/sub-cm4_ses-01_ieeg.json (hunk 8 above)
Comment (choldgraf) [reply ieeg_filtered_speech/sub-cm4/ses-01/sub-cm4_ses-01_ieeg.json]: yep - these are changes I need to add to MNE-BIDS, work on that in the PR here:

https://github.com/mne-tools/mne-bids/pull/28

File: ieeg_filtered_speech/sub-cm8/ses-01/.ipynb_checkpoints/sub-cm8_ses-01_ieeg-checkpoint.json (hunk 9 above)
Comment (choldgraf) [reply ieeg_filtered_speech/sub-cm8/ses-01/.ipynb_checkpoints/sub-cm8_ses-01_ieeg-checkpoint.json]: accidental addition - it's a jupyter auto-generated thing to retain state of your session...will remove in next push

**Comment by choldgraf** [comment]
I _think_ these are addressed and I also just pushed another commit to the mne-bids PR...will take another pass tomorrow when I work w/ alex+mainak

**Comment by choldgraf** [comment]
@dorahermes ready for merge on this one from my perspective. if it turns out we need to change something I can spot-check it with a PR later on. Are you happy w/ the state of this dataset?

**Comment by choldgraf** [comment]
@dorahermes can you give a 👍 or a 👎 on whether we can merge this?

**Comment by dorahermes** [comment]
oh sorry, missed the message on April 23rd... Yes,  👍it passed the validator, correct?

**Comment by robertoostenveld** [comment]
Is there already an ieeg validator? If not, I can branch one, just like I did for EEG at  https://github.com/robertoostenveld/bids-validator/tree/eeg-validator. I can add you as collaborators to that repo so that we can jointly work on the validator (prior to it being merged). This will facilitate also the example development, because once a validator is available somewhere on github, we can use it in travis and cross-validate the examples and the validator.

**Comment by choldgraf** [comment]
@robertoostenveld yep, there's a PR open here: https://github.com/INCF/bids-validator/pull/451

it's mostly ready to go, just need to make a few small changes and merge once the specification is frozen

**Comment by choldgraf** [comment]
merging this, and I can open a PR to update the datasets for changes to the validator later on

**Pull request merged**